app.rb

```diff
@@ -0,0 +1,179 @@
+      list_of_books
+    when '2'
+      list_of_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_of_rentals
+    else
+      'Invalid Input. Enter digit from 1 to 7'
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]'
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'invalid input. Input either 1 or 2 according to your choice'
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books.push(book)
+
+    puts 'Book has been successfully created'
+    go_back
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N] '
+    has_permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(2, age, name, has_permission)
+    @people.push(student)
+
+    puts 'Student has been successfully created'
+    go_back
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'specialization '
+    specialization = gets.chomp.downcase
+
+    teacher = Teacher.new(specialization, age, name)
+    @people.push(teacher)
+
+    puts 'Teacher has been created successfully'
+    go_back
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number (not id)'
+    list_of_books
+    book_index = gets.chomp.to_i
+    book = @books[book_index]
+
+    puts 'Select a person from the following list by number (not id)'
+    list_of_people
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
+
+    print 'Date :'
+    date = gets.chomp
+    actual_date = Date.parse(date)
+
+    rental = Rental.new(actual_date, book, person)
+    @rentals.push(rental)
+
+    puts 'Rental has been successfully created'
+    go_back
+  end
+
+  def list_of_people
+    if @people.empty?
+      puts 'No people found!'
+      go_back
+    end
+
+    @people.each_with_index do |person, i|
+      puts "#{i})  Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
+    end
+  end
+
+  def list_of_books
+    if @books.empty?
+      puts 'No Book found!'
+      go_back
+    end
+
+    @books.each_with_index do |book, i|
+      puts "#{i}) Title: #{book.title}, Author: #{book.author} "
+    end
+  end
+
+  def list_of_rentals
+    puts 'Select id of any person'
+    @people.each { |person| puts "id: #{person.id}, Person: #{person.name}" }
+
+    print 'Person id: '
+    person_id = gets.chomp.to_i
+
+    @rentals.each do |rental|
+      if rental.person.id.to_i == person_id
+        puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      end
+    end
+  end
```

Comment:
- [ ] Your school library console app must behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) it was provided for you. Great job so far 👏🏻From Steps one up to five works fine, however, the last one step six has some mistakes that need to be addressed. Here is the error message I get when I list all rentals for a given person id 👇🏻 Please correct this.

![Screenshot from 2023-01-07 19-20-21](https://user-images.githubusercontent.com/79058364/211162961-8c853a62-2078-4d19-b0e6-62f8ed60a122.png)